core/src/string/interner.rs

```diff
@@ -52,6 +52,9 @@ impl AvmAtom<'_> {
 pub struct AvmStringInterner<'gc> {
     interned: WeakSet<'gc, AvmStringRepr<'gc>>,
 
+    /// Strings used across both AVMs and in core code.
+    pub common: Gc<'gc, CommonStrings<'gc>>,
```

Comment:
Couldn't this stay a `CommonStrings`, without a Gc (and have `common()` return the reference)?
This would mean one less pointer hop.

I just tried making this change and it almost went perfectly smoothly, in `op_type_of` to satisfy the borrow checker I needed to access them directly, without storing `common_strings` in a variable. (alternatively, we can add `self.common_strings()` that returns a non-mut reference, and same with `self.avm2_nomut()` - since we don't actually need a `&mut` anywhere there.)